rts/Sim/MoveTypes/HoverAirMoveType.cpp

```diff
@@ -410,7 +410,7 @@ void CHoverAirMoveType::UpdateHovering()
 
 	// randomly drift (but not too far from goal-position)
 	wantedSpeed = (randomWind * math::fabs(owner->unitDef->dlHoverFactor) * 0.5f);
-	wantedSpeed += (smoothstep(0.0f, 20.0f * 20.0f, (goalPos - owner->pos)) * (goalPos - owner->pos));
+	wantedSpeed += (goalPos - owner->pos) * 0.1f;
```

Comment:
Ok, I see. If the randomWind would blow it past goalPos, this nudges it even further past it.

The problem with the smoothstep is that it applies a drift into quadrant I. It never pushes the unit back into -X-Y.